app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/service.tsx

```diff
@@ -10,20 +10,107 @@ import * as loggerProvider from "../providers/logger";
 // === Constants ===
 // =================
 
+/** Default HTTP body for an "open project" request. */
+const DEFAULT_OPEN_PROJECT_BODY: OpenProjectRequestBody = {
+  forceCreate: false,
+};
+
 /** Relative HTTP path to the "get user" endpoint of the Cloud backend API. */
 const GET_USER_PATH = "users/me";
+/** Relative HTTP path to the "list projects" endpoint of the Cloud backend API. */
+const LIST_PROJECTS_PATH = "projects";
+/** Relative HTTP path to the "create project" endpoint of the Cloud backend API. */
+const CREATE_PROJECT_PATH = "projects";
+/** Relative HTTP path to the "close project" endpoint of the Cloud backend API. */
+const closeProjectPath = (projectId: ProjectId) =>
+  `projects/${projectId}/close`;
+/** Relative HTTP path to the "get project" endpoint of the Cloud backend API. */
+const getProjectPath = (projectId: ProjectId) => `projects/${projectId}`;
+/** Relative HTTP path to the "open project" endpoint of the Cloud backend API. */
+const openProjectPath = (projectId: ProjectId) => `projects/${projectId}/open`;
 
 // =============
 // === Types ===
 // =============
 
+/** Unique identifier for a user's project. */
+export type ProjectId = string;
+
 /** A user/organization in the application. These are the primary owners of a project. */
 export interface Organization {
   id: string;
   userEmail: string;
   name: string;
 }
 
+/** Type of application that a {@link Version} applies to.
+ *
+ * We keep track of both backend and IDE versions, so that we can update the two independently.
+ * However the format of the version numbers is the same for both, so we can use the same type for
+ * both. We just need this enum to disambiguate. */
+export enum VersionType {
+  backend = "Backend",
+  ide = "Ide",
+}
+
+/** A version describing a release of the backend or IDE. */
+export interface Version {
```

Comment:
this type definition may be incorrect ([see link above](https://github.com/enso-org/enso/blob/5fb66bb0679db2a62b8f43398e565f2e2e708ba9/app/ide-desktop/lib/dashboard/src/authentication/src/dashboard/service.tsx))

Reply:
No longer relevant since these types were refactored out to a future PR.